### PR TITLE
Fix encoded urls when using phantomjs 2.1.0 or newer

### LINKED ISSUE #1464: Encoded urls fail when using phantomjs 2.2.1
If the first url(s) opened requires encoding, the response lacks the status. After a url is opened that does not require encoding, urls that require encoding work okay. In the following sequence, url 1 fails, url 2 works, url 3 works:

    page%20_1.html
    page_2.html
    page%20_3.html

In this sequence, all work okay:

    page_1.html
    page%20_2.html
    page%20_3.html

Stay tuned for a PR with a test that demonstrates the problem.

### PR DESCRIPTION
**Pull request by mepard**
Fixes #1464. Rebased #1465.

### CODE REVIEW

File: modules/casper.js [hunk 1]
```diff
@@ -2637,7 +2639,8 @@ function createPage(casper) {
     };
     page.onResourceRequested = function onResourceRequested(requestData, request) {
         casper.emit('resource.requested', requestData, request);
-        if (requestData.url === casper.requestUrl) {
+    	var checkUrl = utils.ltVersion(phantom.version, '2.1.0') ? utils.decodeUrl(requestData.url) : requestData.url;
```
Comment (istr) [inline modules/casper.js]: whitespace / lint

**Comment by istr** [comment]
@mickaelandrieu @BIGjuevos I would like to merge this. Please have a look. Thank you.

File: modules/casper.js [hunk 2]
```diff
@@ -1209,7 +1209,9 @@ Casper.prototype.handleReceivedResource = function(resource) {
         return;
     }
     this.resources.push(resource);
-    if (utils.decodeUrl(resource.url) !== this.requestUrl) {
+
+    var checkUrl = utils.ltVersion(phantom.version, '2.1.0') ? utils.decodeUrl(resource.url) : resource.url;
```
Comment (BIGjuevos) [inline modules/casper.js]: Hoist your variables?

**Comment by BIGjuevos** [comment]
No objections to the code overall, just some minor style updates and it get's my :+1:

Thanks for your hard work @mepard

**Comment by istr** [comment]
We should keep this in mind and maybe put that code snippet in a helper function. This pattern occurs in other places, see e.g. #1417.

**Comment by mickaelandrieu** [comment]
:+1:  ok too :)

regarding your last comment @istr, I agree, let's see how this pattern evolve and we will refactor when required, what do you think ?

**Comment by istr** [comment]
@mickaelandrieu 
> regarding your last comment @istr, I agree, let's see how this pattern evolve and we will refactor when required

Absolutely. Let us land this now and refactor when necessary.

**Comment by istr** [comment]
Merging because it fixes a real bug and has a very useful test-case. Postpone linting / refactoring.

**Pull request merged**